harness/determined/pytorch/_callback.py

```diff
@@ -51,6 +51,18 @@ def on_validation_step_end(self, metrics: Dict[str, Any]) -> None:
         # TODO(DET-3555): remove this once it has been deprecated long enough.
         pass
 
+    def on_checkpoint_before_load(self, checkpoint: Dict[str, Any]) -> None:
```

Comment:
it might be more consistent to use name like `on_checkpoint_load_start` and `on_checkpoint_save_start` but I am bad at names.